packages/test/functional-tests/src/test/ddsHandleEncoding.spec.ts

```diff
@@ -102,25 +110,72 @@ describe("DDS Handle Encoding", () => {
 			},
 			[handle.absolutePath] /* expectedHandles */,
 		),
+		createTestCase(
+			new SharedStringFactory(),
+			(dds) => {
+				dds.insertMarker(0, ReferenceType.Simple, { marker: handle });
+			},
+			[handle.absolutePath] /* expectedHandles */,
+		),
+		createTestCase(
+			new SharedMatrixFactory(),
+			(dds: SharedMatrix) => {
+				dds.insertRows(0, 1);
+				dds.insertCols(0, 1);
+
+				dds.setCell(0, 0, handle);
+			},
+			[handle.absolutePath] /* expectedHandles */,
+		),
+		createTestCase(
+			new TreeFactory({}),
+			(dds: ITree) => {
+				const builder = new SchemaFactory("test");
+				class Bar extends builder.object("bar", {
+					h: builder.optional(builder.handle),
+				}) {}
+
+				const config = new TreeConfiguration(Bar, () => ({
+					h: undefined,
+				}));
+
+				const treeView = dds.schematize(config);
+
+				treeView.root.h = handle;
+			},
+			[handle.absolutePath] /* expectedHandles */,
+		),
 		createTestCase(
 			new ConsensusRegisterCollectionFactory(),
 			(dds) => {
-				dds.write("whatever", handle).catch(() => {
-					// We only care about errors before message submission, which will fail the message.length assert below.
-				});
+				dds.write("whatever", handle).catch(() => {});
 			},
-			[] /* expectedHandles */,
+			[handle.absolutePath] /* expectedHandles */,
+		),
+		createTestCase(
+			new ConsensusQueueFactory(),
+			(dds) => {
+				dds.add(handle).catch(() => {});
+			},
+			[handle.absolutePath] /* expectedHandles */,
```

Comment:
I thought this one wasn't going to work.  I haven't looked at the code yet though...